### PR TITLE
reference 9525, say subjectDN for EE should be null, close #35

### LINKED ISSUE #35: Subject field
> RFC5280 defines the Subject field as follows: "The subject field identifies the entity associated with the public key stored in the subject public key field." RFC 5280 adds "If the subject is a CA then the subject field MUST be populated with a non-empty distinguished name matching the contents of the issuer field in all certificates issued by the subject CA."

Michael: Some reference to UTA-rfc6125bis seems in order to say that CN= in the DN is not desired, but since IssuerDN can not empty, we need something.

Secondly, do you want to reference the PKI level text from draft-irtf-t2trg-taxonomy-manufacturer-anchors-02

> The Subject field MUST be present and MUST contain the commonName, the organizationName, and the countryName attribute and MAY contain an organizationalUnitName attribute.

Michael: Leave out CN, contra-indicated by 6125bis.

Thomas: +1. And more generally we should look carefully at 6125bis and make sure we are aligned. See delta: https://www.ietf.org/archive/id/draft-ietf-uta-rfc6125bis-15.html#name-changes-from-rfc-6125

### PR DESCRIPTION
**Pull request by mcr**
close #35

### CODE REVIEW

File: draft-ietf-uta-tls13-iot-profile.md [hunk 1]
```diff
@@ -489,8 +489,9 @@ field." RFC 5280 adds "If the subject is a CA then the subject field MUST be
 populated with a non-empty distinguished name matching the contents of the
 issuer field in all certificates issued by the subject CA."
 
-The Subject field MUST be present and MUST contain the commonName, the organizationName,
-and the countryName attribute and MAY contain an organizationalUnitName attribute.
+However, {{!RFC9525, Section 2.9}} now recommends that the SubjectDN be empty (null) for all End Entity certificates.
+
+Root CA and Subordinate CAs must have a non-null SubjectDN as that value must match the IssuerDN of subordinate certificates.
```
Comment (hannestschofenig) [inline draft-ietf-uta-tls13-iot-profile.md:494]:
```suggestion
Root CA certificates and Subordinate CA certificates MUST have a non-null SubjectDN, as the value MUST match the DN of the Issuer.
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 2]
```diff
@@ -489,8 +489,9 @@ field." RFC 5280 adds "If the subject is a CA then the subject field MUST be
 populated with a non-empty distinguished name matching the contents of the
 issuer field in all certificates issued by the subject CA."
 
-The Subject field MUST be present and MUST contain the commonName, the organizationName,
-and the countryName attribute and MAY contain an organizationalUnitName attribute.
+However, {{!RFC9525, Section 2.9}} now recommends that the SubjectDN be empty (null) for all End Entity certificates.
```
Comment (mcr) [inline draft-ietf-uta-tls13-iot-profile.md:492]:
```suggestion
However, as {{!RFC9525, Section 2}} mandates that the subjectDN not be be used to identify a service, for IoT purposes, an empty SubjectDN avoids all confusion for End Entity certificates.
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 3]
```diff
@@ -489,7 +489,9 @@ field." RFC 5280 adds "If the subject is a CA then the subject field MUST be
 populated with a non-empty distinguished name matching the contents of the
 issuer field in all certificates issued by the subject CA."
 
-The Subject field MUST be present and MUST contain the commonName, the organizationName,
+Root CA certificates and Subordinate CA certificates MUST have a non-empty SubjectDN, as the value MUST match the DN of the Issuer.
```
Comment (mcr) [inline draft-ietf-uta-tls13-iot-profile.md:492]:
```suggestion
Root CA certificates MUST have a non-empty SubjectDN.
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 4]
```diff
@@ -481,16 +481,17 @@ MUST NOT be marked critical.
...
-{{!RFC5280}} defines the Subject field as follows: "The subject field identifies
+{{!RFC5280}} defines the subjectName field as follows: "The subject field identifies
 the entity associated with the public key stored in the subject public key
 field." RFC 5280 adds "If the subject is a CA then the subject field MUST be
 populated with a non-empty distinguished name matching the contents of the
 issuer field in all certificates issued by the subject CA."
 
-The Subject field MUST be present and MUST contain the commonName, the organizationName,
-and the countryName attribute and MAY contain an organizationalUnitName attribute.
+Root CA certificates MUST have a non-empty SubjectDN.
```
Comment (mcr) [inline draft-ietf-uta-tls13-iot-profile.md:492]:
```suggestion
Root CA certificates MUST have a non-empty subjectName.
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 5]
```diff
@@ -565,9 +558,9 @@ be set to true and the pathLenConstraint MUST be omitted.
 
 This section outlines the requirements for subordinate CA certificates.
 
-### Subject
+### subjectName
 
-The Subject field MUST be set and MUST contain the commonName, the organizationName,
+The subjectName field MUST be set and MUST contain the commonName, the organizationName,
```
Comment (hannestschofenig) [inline draft-ietf-uta-tls13-iot-profile.md:563]:
```suggestion
The subject field MUST be set and MUST contain the commonName, the organizationName,
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 6]
```diff
@@ -615,18 +608,20 @@ status service (OCSP).
...
+entity certificates as a subjectName is lifted.
 
 Two fields are typically used to encode a device identifer, namely the
 Subject and the subjectAltName fields. Protocol specifications tend to offer
 recommendations what identifiers to use and the deployment situation is
 fragmented.
 
-The Subject field MAY include a unique device serial number. If the serial
-number is included, it MUST be encoded in the serialNumber attribute.
+The subjectName field MAY include a unique device serial number. If the serial
```
Comment (hannestschofenig) [inline draft-ietf-uta-tls13-iot-profile.md:623]:
```suggestion
The subject field MAY include a unique device serial number. If the serial
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 7]
```diff
@@ -615,18 +608,20 @@ status service (OCSP).
...
 This section outlines the requirements for end entity certificates.
 
-### Subject
+### subjectName
+
+{{!RFC9525, Section 2}} mandates that the subjectName not be be used to identify a service, for IoT purposes, an empty subjectName avoids all confusion for End Entity certificates.
 
 The requirement in Section 4.4.2 of {{!RFC7925}} to only use EUI-64 for end
-entity certificates as a Subject name is lifted.
+entity certificates as a subjectName is lifted.
```
Comment (hannestschofenig) [inline draft-ietf-uta-tls13-iot-profile.md:616]:
```suggestion
entity certificates as a subject field is lifted.
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 8]
```diff
@@ -565,9 +558,9 @@ be set to true and the pathLenConstraint MUST be omitted.
 
 This section outlines the requirements for subordinate CA certificates.
 
-### Subject
+### subjectName
```
Comment (hannestschofenig) [inline draft-ietf-uta-tls13-iot-profile.md:561]:
```suggestion
### Subject
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 9]
```diff
@@ -481,16 +481,9 @@ MUST NOT be marked critical.
 
 This section outlines the requirements for root CA certificates.
 
-### Subject
+## Subject
```
Comment (thomas-fossati) [inline draft-ietf-uta-tls13-iot-profile.md:484]:
```suggestion
### Subject
```

Comment (thomas-fossati) [reply draft-ietf-uta-tls13-iot-profile.md:484]: need to increase indent

File: draft-ietf-uta-tls13-iot-profile.md [hunk 10]
```diff
@@ -615,18 +608,20 @@ status service (OCSP).
 
 This section outlines the requirements for end entity certificates.
 
-### Subject
+### subjectName
```
Comment (thomas-fossati) [inline draft-ietf-uta-tls13-iot-profile.md:611]:
```suggestion
### Subject
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 11]
```diff
@@ -617,16 +610,18 @@ This section outlines the requirements for end entity certificates.
 
 ### Subject
 
+{{!RFC9525, Section 2}} mandates that the subject field not be be used to identify a service, for IoT purposes, an empty subjectName avoids all confusion for End Entity certificates.
```
Comment (thomas-fossati) [inline draft-ietf-uta-tls13-iot-profile.md:613]:
```suggestion
{{!RFC9525, Section 2}} mandates that the subject field not be used to identify a service.
For IoT purposes, an empty subject field avoids all confusion for End Entity certificates.
```

**Review by thomas-fossati — APPROVED** [review]
LGTM

**Pull request merged**